### PR TITLE
Fix OPT doc tests

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes the doctest for OPT by adding slight modifications.

### CODE REVIEW

**Comment by ydshieh** [comment]
Hi @ArthurZucker Do you know why the expected value has extra ``` ? `OPTForCausalLM` has the same expected value, but it doesn't have test failure. Seems strange to me 😕 

```
 1003         >>> # Generate
 1004         >>> generate_ids = model.generate(inputs.input_ids, max_length=30)
 1005         >>> tokenizer.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]
 Expected:
     "Hey, are you consciours? Can you talk to me?\nI'm not consciours, but I can talk to you."
     ```
 Got:
     "Hey, are you consciours? Can you talk to me?\nI'm not consciours, but I can talk to you."
```

**Comment by ArthurZucker** [comment]
Actually it seems that the automatic doc préparation that adds a line does not affect the TFOPTForCausalLM forward doc. Thus I removed the manually created example and defeated to using **add_code_sample** which worked in that case. 
Now why does the add extra line does not work is another mystery and I could not setup a debugging instance to check that

**Comment by ydshieh** [comment]
Thank you. I will check this on Monday.

File: src/transformers/models/opt/modeling_tf_opt.py [hunk 1]
```diff
@@ -53,6 +53,9 @@
 # Base model docstring
 _EXPECTED_OUTPUT_SHAPE = [1, 8, 1024]
 
+# Causal LM output
+_EXPECTED_CAUSAL_LM_OUTPUT = "Hey, are you consciours? Can you talk to me?\nI'm not consciours, but I can talk to you."
```
Comment (ydshieh) [inline src/transformers/models/opt/modeling_tf_opt.py:57]: Maybe `_CAUSAL_LM_EXPECTED_OUTPUT` to keep aligned with other names?

**Review by ydshieh — APPROVED** [review]
Thank you for the fix, @ArthurZucker ! LGTM, especially all tests pass now.
Let's wait an approval from some core maintainers.

File: src/transformers/models/opt/modeling_tf_opt.py (hunk 1 above)
Comment (ArthurZucker) [reply src/transformers/models/opt/modeling_tf_opt.py:57]: Good catch

**Comment by ydshieh** [comment]
Regarding the mystery , it is due to this line

> ``decoder_input_ids```

https://github.com/ArthurZucker/transformers/blob/c1950186fba25d0219db0db7a27e9f0a0b7225c0/src/transformers/models/opt/modeling_tf_opt.py#L604

**Comment by ArthurZucker** [comment]
I forgot to change the name in the `add_docstring` will fix that

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**